src/error-snippets/source-maps.ts

```diff
@@ -0,0 +1,45 @@
+import { SourceMapConsumer, type BasicSourceMapConsumer } from "source-map";
+import url from "url";
+import { SOURCE_MAP_URL_COMMENT } from "./constants";
+import { softFileURLToPath, getSourceCodeFile } from "./utils";
+import type { SufficientStackFrame, ResolvedFrame } from "./types";
+
+export const extractSourceMaps = async (
+    fileContents: string,
+    fileName: string,
+): Promise<BasicSourceMapConsumer | null> => {
+    const sourceMapsStartIndex = fileContents.indexOf(SOURCE_MAP_URL_COMMENT);
+    const sourceMapsEndIndex = fileContents.indexOf("\n", sourceMapsStartIndex);
+
+    if (sourceMapsStartIndex === -1) {
```

Comment:
so this check needs to be moved higher

Reply:
Yes, we could save some CPU cycles with it (and nothing more), but its more readable:
```ts
    const sourceMapsStartIndex = fileContents.indexOf(SOURCE_MAP_URL_COMMENT);
    const sourceMapsEndIndex = fileContents.indexOf("\n", sourceMapsStartIndex);

    if (sourceMapsStartIndex === -1) {
        return null;
    }
```
```ts
    const sourceMapsStartIndex = fileContents.indexOf(SOURCE_MAP_URL_COMMENT);

    if (sourceMapsStartIndex === -1) {
        return null;
    }

    const sourceMapsEndIndex = fileContents.indexOf("\n", sourceMapsStartIndex);
```